buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -588,7 +588,7 @@ static ByteBuf encodeString0(ByteBufAllocator alloc, boolean enforceHeap, CharBu
             dst = alloc.buffer(length);
         }
         try {
-            final ByteBuffer dstBuf = dst.internalNioBuffer(0, length);
+            final ByteBuffer dstBuf = dst.internalNioBuffer(dst.readerIndex(), length);
```

Comment:
If the allocator always returns a new buffer and the reader index is guaranteed to be 0 ... then it makes no difference. However I'm not sure we need to make those assumptions in this code.